lib/compress/zstdmt_compress.c

```diff
@@ -683,23 +720,22 @@ static void ZSTDMT_waitForAllJobsCompleted(ZSTDMT_CCtx* zcs)
     }
 }
 
-
-/** ZSTDMT_initCStream_internal() :
- *  internal usage only */
-size_t ZSTDMT_initCStream_internal(ZSTDMT_CCtx* zcs,
-                    const void* dict, size_t dictSize, const ZSTD_CDict* cdict,
-                    ZSTD_parameters params, unsigned long long pledgedSrcSize)
+size_t ZSTDMT_initCStream_internal(
+        ZSTDMT_CCtx* zcs, const void* dict, size_t dictSize,
+        const ZSTD_CDict* cdict, ZSTD_CCtx_params cctxParams,
+        unsigned long long pledgedSrcSize)
```

Comment:
This leads me to a question : to we want `pledgedSrcSize` outside or within `cctxParams` ?

Reply:
It is a parameter, albeit maybe a "single use" one. When calling `ZSTD_CCtx_reset`, we would probably still want to reset it, which might be strange since all the other parameters persist?

Reply:
Yes, that's a good point.
It's indeed only valid for the current session.
So let's keep it outside of `ZSTD_CCtx_params`, since its lifetime is different.